### PR TITLE
Allow flake8 to ignore certain directories

### LINKED ISSUE #751: Make sure flake8 checks only relevant files
From https://github.com/fairlearn/fairlearn/pull/747#issuecomment-821531625:

> Sadly this PR can't be completed since I managed to create a branch with `.py` at the end which means the `.git` folder has a log file ending in `.py` and `flake8` feels responsible for linting it, which results in errors. Wow! Will create a new branch with identical content and better name. Closing this PR

We need to change our CI to make sure `flake8` only checks the relevant files, and not the whole repo folder.

cc @romanlutz

### PR DESCRIPTION
**Pull request by bthng**
CI should be allowed to ignore certain directories.

Closes #751

### CODE REVIEW

**Review by romanlutz — APPROVED** [review]
Thanks! I'm glad this is getting fixed 🙂 @riedgar-ms or @hildeweerts please take a look and merge if you approve.

File: setup.cfg [hunk 1]
```diff
@@ -8,7 +8,10 @@ copyright-regexp = # Copyright \(c\) Microsoft Corporation and Fairlearn contrib
 # An update to pip has resulted in `pip install .` leaving a 'build' directory around
 # Also: 'pip freeze' does not report the pip version by default
 exclude =
+    .git/
     build/
+    docs/
```
Comment (riedgar-ms) [inline setup.cfg:13]: We probably should keep `flake8` running on `docs/conf.py`.

Comment (bthng) [reply setup.cfg]: That is true, I replaced `docs/` with `docs/auto_examples` instead. Would this work?

**Review by riedgar-ms — APPROVED** [review]
Thanks!

File: setup.cfg (hunk 1 above)
Comment (romanlutz) [reply setup.cfg]: That should do it!

**Comment by bthng** [comment]
The `MacOS Unit 3.7` job seems to be experiencing a timeout [here](https://dev.azure.com/responsibleai/fairlearn/_build/results?buildId=11778&view=logs&j=0a22f5d8-9f3e-5735-4672-5a55fb1a2abf), not too sure how to resolve it

**Comment by romanlutz** [comment]
Thanks again @bthng ! Making the world/repo a better place one commit at a time 😄

**Pull request merged**